docs/guides/themeable-to-emotion-migration-guide.md

```diff
@@ -0,0 +1,459 @@
+
+```js
+// before
+import { themeable } from '@instructure/ui-themeable'
+import theme from './theme'
+import styles from './styles.css'
+
+// after
+import { withStyle, jsx } from '@instructure/emotion'
+import generateComponentTheme from './theme'
+import generateStyle from './styles'
+```
+
+##### 3. decorator
+
+Replace `@themeable` decorator with `@withStyle(generateStyle, generateComponentTheme)`, passing the style and theme generators. (If needed, you can pass `null` instead of both, e.g. when there is a stylesheet, but you don't use any theme variables.)
+
+```js
+// before
+@themeable(theme, styles)
+class ExampleComponent extends Component { ... }
+
+// after
+@withStyle(generateStyle, generateComponentTheme)
+class ExampleComponent extends Component { ... }
+```
+
+##### 4. props and lifecycle methods
+
+The props `makeStyles` and `styles` are added by the decorator, so you might run into eslint errors when using them in the component. We recommend adding them to propTypes, and suppressing the eslint errors there.
+
+In the `componentDidMount` and `componentDidUpdate` methods, call the `makeStyles` method (available on this.props) to generate the styles object, passing the state (or any other object needed).
+
+```js
+static propTypes = {
+  // eslint-disable-next-line react/require-default-props
+  makeStyles: PropTypes.func,
+  // eslint-disable-next-line react/require-default-props
+  styles: PropTypes.object,
+  ...
+}
+
+componentDidMount() {
+  this.props.makeStyles({ focused: this.state.focused })
+}
+
+componentDidUpdate() {
+  this.props.makeStyles({ focused: this.state.focused })
+}
+```
+
+##### 5. render and applying styles
+
+In the `render` method, use emotion's `css={this.props.styles.componentName}` syntax to add styles. Refactor your code to move all style-related logic to the `styles.js`.
+
+Before:
+
+```jsx
+// before in index.js
+render() {
+  const classes = {
+    [styles.root]: true,
+    [styles[size]]: true,
+    [styles.disabled]: disabled
+  }
+
+  return (
+    <div className={classnames(classes)}>...</div>
+  )
+}
+```
+
+```css
+/* before in styles.css */
+.root {
+  display: block;
+  font-family: var(--fontFamily);
+}
+.small {
+  font-size: 1em;
+}
+.medium {
+  font-size: 1.5em;
+}
+.large {
+  font-size: 2em;
+}
+.disabled {
+  pointer-events: none;
+}
+```
+
+After:
+
+```jsx
+// after in index.js
+render() {
+  return (
+    <div css={styles.componentName}>...</div>
+  )
+}
+```
+
+```js
+// after in styles.js
+const sizeVariants = {
+  small: { fontSize: '1em' },
+  medium: { fontSize: '1.5em' },
+  large: { fontSize: '2em' }
+}
+
+return {
+  componentName: {
+    label: 'componentName',
+    display: 'block',
+    fontFamily: componentTheme.fontFamily,
+    ...sizeVariants[props.size],
+    ...(props.disabled && { pointerEvent: 'none' })
+  }
+}
+```
+
+#### Example Component
```

Comment:
```suggestion
#### The completed example component
```

Reply:
Changed it to "An example component using emotion theming". "Completed" would be valid if it was the sum of the previous examples stiched together, but it's not completely that.